### PR TITLE
[d3-array] Update types to v2.4

### PR DESCRIPTION
**Pull request by Methuselah96**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/d3/d3-array/releases/tag/v2.4.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

Related: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/38939

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## d3-array ([<kbd>unpkg</kbd>](https://unpkg.com/browse/d3-array@latest/))
was missing the following properties:
1. scan
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 80fe59cd36a8f42bd3318f7ab7098f4cea9a4bdb
</p>

**Comment by Methuselah96** [comment]
@RyanCavanaugh Can I get a review? I'm trying get through [a lot of PRs](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/38939#issuecomment-696772770) and there aren't very active maintainers for these types. orta [suggested](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/48542#issuecomment-710048882) I batch them so that it would get on the "Waiting for Code Reviews" list, but it looks like elizbarzilay moved it to "Waiting for Code Reviews" anyway.

File: types/d3-array/index.d.ts
Comment (Ledragon) [inline types/d3-array/index.d.ts:183]: Shouldn't `T`  be `T|undefined|null`?

Comment (Methuselah96) [reply types/d3-array/index.d.ts:183]: I didn't think about it too much since I just used followed the pattern of the already existing `sum`, `mean`, `median`, `quantile`, etc. However I think it's acceptable to just be `T`, since `T` can include `| undefined | null` if desired and it's really the output value that we care about/know about. We don't know anything about `T`.

Comment (Ledragon) [reply types/d3-array/index.d.ts:183]: Ok, fine for me :-)

**Review by Ledragon — APPROVED** [review]

**Comment by Methuselah96** [comment]
Ready to merge

**Pull request merged**
